File: test/test_strscan.rb
# need complete path
require '/Users/remy/Github/dyndoc-opal/test/dyndoc_helpers.rb'

#s,r,r2="tototito",/tot/,/ti/
s,r,r2="{#document][#main]titi[#}",/\{\#/,/(\]?)\s*(\[[\#\@]([\w\:\|-]*[<>]?[=?!><]?)\})/
a=DyndocStringScanner.new(s)

p (a.exist? r)

p [a.string,a.pos]

p "scan"

p (a.scan r)

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

a.reset

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

a.pos=18
p "scan_until"

p (a.scan_until r2)

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]
p [:match,a.match] if RUBY_ENGINE=="opal"

a.reset

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

p "check"

p (a.check r)

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

a.reset

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

p "check_until"

p (a.check_until r2)

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

a.reset

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

p "skip"

p (a.skip r)

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

a.reset

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

p "skip_until"

p (a.skip_until r2)

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]

a.reset

p [a.string,a.pos,a.rest,a.pre_match,a.matched,a.post_match]


# txt = ""

# txt += "toto"

# p txt

# p txt.byteslice 0,3

# p txt.byteslice 0...3

File: test/test_strscan.js
var Opal=require("../dist/npm/dyndoc.js")()
var Dyndoc=Opal.Dyndoc


var s = "{#document][#main]titi[#}"
var r=/\{\#/,r2=/(\]?)\s*(\[[\#\@]([\w\:\|-]*[<>]?[=?!><]?)\})/

var sscan = Opal.DyndocStringScanner.$new(s)

console.log(sscan)


console.log(sscan["$exist?"](r))

console.log(sscan["$exist?"](/to/))


console.log("scan")

console.log(sscan.$scan(r))

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])

sscan.$reset()

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])

sscan["$pos="](18)
console.log("scan_until")

console.log(sscan.$scan_until(r2))

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])
console.log(["match",sscan.match])

sscan.$reset()

console.log("check")

console.log(sscan.$check(r))

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])

sscan.$reset()

console.log("check_until")

console.log(sscan.$check_until(r2))

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])

sscan.$reset()

console.log("skip")

console.log(sscan.$skip(r))

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])

sscan.$reset()


console.log("skip_until")

console.log(sscan.$skip_until(r2))

console.log([sscan.$string(),sscan.$pos(),sscan.$rest(),sscan.$pre_match(),sscan.$matched(),sscan.$post_match()])
